### ISSUE TITLE
NPM_AUTH is not set but required for project specific .npmrc files

### ISSUE THREAD

**Issue by adborden** [goodeggs/developer-start#30]
Something like this in .profile would probabably work (until npm 2).

```
if [[ -f ~/.npmrc ]]; then
  NPM_AUTH=$(grep _auth ~/.npmrc | sed -e 's/^_auth=//') ;  export NPM_AUTH
fi
```

**Comment by randypuro** [comment]
I think I fixed this:

https://github.com/goodeggs/developer-start/issues/29

**Comment by adborden** [comment]
Ah! I never looked at the script! Your subshell isn't quoted, so the .profile looks like this on disk
:
```
# project specific .npmrc files require NPM_AUTH environment (as used in CI)
if [ -f ~/.npmrc ]; then
  export NPM_AUTH=
fi
```

**Comment by randypuro** [comment]
Ha, that's a terrible fix then. I thought I tried it locally setting up my new machine, but looking in my .profile, I see that it's just exporting the current value.

Just pushed the fix.

**Comment by adborden** [comment]
Bummer, just found another issue. The token is base64, so it may contain "=" which awk will treat as a field separator, thus truncating the token.

**Comment by randypuro** [comment]
Of course, this is what I get for copying things off the internet. The
toughest few line fix evveeer.

On Thu, Jun 18, 2015 at 5:42 PM, Aaron Borden <notifications@github.com>
wrote:

> Bummer, just found another issue. The token is base64, so it may contain
> "=" which awk will treat as a field separator, thus truncating the token.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/goodeggs/developer-start/issues/30#issuecomment-113329836>
> .
>



-- 
randy puro || good eggs || 510.326.2789

dig in: twitter <http://twitter.com/goodeggs> || facebook
<http://facebook.com/goodeggs> || blog <http://blog.goodeggs.com/>

**Comment by randypuro** [comment]
Did you see it fail Aaron? I stuck another "=" in mine, and it still worked.

On Fri, Jun 19, 2015 at 10:25 AM, Randy Puro <randy@goodeggs.com> wrote:

> Of course, this is what I get for copying things off the internet. The
> toughest few line fix evveeer.
>
> On Thu, Jun 18, 2015 at 5:42 PM, Aaron Borden <notifications@github.com>
> wrote:
>
>> Bummer, just found another issue. The token is base64, so it may contain
>> "=" which awk will treat as a field separator, thus truncating the token.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/goodeggs/developer-start/issues/30#issuecomment-113329836>
>> .
>>
>
>
>
> --
> randy puro || good eggs || 510.326.2789
>
> dig in: twitter <http://twitter.com/goodeggs> || facebook
> <http://facebook.com/goodeggs> || blog <http://blog.goodeggs.com/>
>



-- 
randy puro || good eggs || 510.326.2789

dig in: twitter <http://twitter.com/goodeggs> || facebook
<http://facebook.com/goodeggs> || blog <http://blog.goodeggs.com/>

**Comment by adborden** [comment]
It doesn't fail with an error, you just get an incomplete token:
.npmrc:
```
_auth="fQ=="
```


    $ export NPM_AUTH=$(awk -F "=" '/_auth/ {print $2}' ~/.npmrc)
    $ echo $NPM_AUTH
        "fQ

I can spend some time on it, there's also some work to do to get ready for NPM2, and the current strategy would need to be tweaked anyway.

**Comment by randypuro** [comment]
Yeah, weird, depends on where the extra '=' is at. I just npm logged in
again (not sure why I had), and it's not handling the quotes correctly
either.

On Fri, Jun 19, 2015 at 11:06 AM, Aaron Borden <notifications@github.com>
wrote:

> It doesn't fail with an error, you just get an incomplete token:
> .npmrc:
>
> _auth="fQ=="
>
> $ export NPM_AUTH=$(awk -F "=" '/_auth/ {print $2}' ~/.npmrc)
> $ echo $NPM_AUTH
>     "fQ
>
> I can spend some time on it, there's also some work to do to get ready for
> NPM2, and the current strategy would need to be tweaked anyway.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/goodeggs/developer-start/issues/30#issuecomment-113593096>
> .
>



-- 
randy puro || good eggs || 510.326.2789

dig in: twitter <http://twitter.com/goodeggs> || facebook
<http://facebook.com/goodeggs> || blog <http://blog.goodeggs.com/>
